File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.cpp
class Solution {
public:
    bool checkOnesSegment(string s) {
        int seg = 0;
        int last = 0;

        for(char ch : s) {
            int c = ch - '0';
            if(c==1) {
                if(last == 0) seg++;
            }
            last = c;
        }

        return seg <= 1;
    }
};

File: 1784-check-if-binary-string-has-at-most-one-segment-of-ones/1784-check-if-binary-string-has-at-most-one-segment-of-ones.go
func checkOnesSegment(s string) bool {
    seg := 0
    last := 0
    for _, ch := range s {
        c := int(byte(ch) - '0')
        if c == 1{
            if last == 0 {
                seg++
            }
        }
        last = c
    }

    return seg <= 1
}